### PR TITLE
bugfix: dataDisplay/CoreTable related CoreComponents prop sanitization

### LINKED ISSUE #336: bug: dataDisplay/CoreTable valid props and sanitization.
**Describe the bug**
Valid props and props sanitization missing:

- [x] 1. CoreTable
- [x] 2. CoreTableBody
- [ ] 3. CoreTableBodyCell
- [x] 4. CoreTableCell
- [x] 5. CoreTableContainer
- [x] 6. CoreTableFooter
- [x] 7. CoreTableHead
- [ ] 8. CoreTableHeadCell
- [ ] 9. CoreTableHeadRow
- [x] 10. CoreTablePagination
- [x] 11. CoreTableRow
- [x] 12. CoreTableSortLabel

### PR DESCRIPTION
**Pull request by anantakumarghosh**
## Description

This PR adds valid props and prop sanitizations for below listed CoreComponents:

- CoreTable
- CoreTableBody
- CoreTableCell
- CoreTableContainer
- CoreTableFooter
- CoreTableHead
- CoreTablePagination
- CoreTableRow
- CoreTableSortLabel

Ref: #336 

## Checklist

- [x] I have performed a thorough self-review of my code.
- [ ] I have added or updated relevant tests for my changes.
- [x] My code follows the project's style guidelines and best practices.
- [ ] I have updated the documentation if necessary.
- [ ] I have added or updated relevant comments in my code.
- [ ] I have resolved any merge conflicts of my branch.

## Reviewers

---

## Maintainer Notes

- [ ] Has this change been tested in a staging environment?
- [ ] Does this change introduce any breaking changes or deprecations?

### CODE REVIEW

File: package/components/dataDisplay/CoreTable.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default function CoreTable(props) {
   } = props;
 
   return (
-    <NativeTable 
+    <NativeTable
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTable.js]: Unnecessary props destructure.

File: package/components/dataDisplay/CoreTableBody.js [hunk 2]
```diff
@@ -4,6 +4,24 @@ import React from "react";
 // eslint-disable-next-line import/no-unresolved
 import { NativeTableBody } from "@wrappid/native";
 
+import { sanitizeComponentProps } from "../../utils/componentUtil";
+
 export default function CoreTableBody(props) {
-  return <NativeTableBody {...props}>{props.children}</NativeTableBody>;
+  props = sanitizeComponentProps(CoreTableBody, props);
+  return <NativeTableBody {...props}>{props?.children}</NativeTableBody>;
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTableBody.js]: Can be self closing.

File: package/components/dataDisplay/CoreTableContainer.js [hunk 3]
```diff
@@ -4,6 +4,25 @@ import React from "react";
 // eslint-disable-next-line import/no-unresolved
 import { NativeTableContainer } from "@wrappid/native";
 
+import { sanitizeComponentProps } from "../../utils/componentUtil";
+
 export default function CoreTableContainer(props) {
-  return <NativeTableContainer {...props}>{props.children}</NativeTableContainer>;
+  props = sanitizeComponentProps(CoreTableContainer, props);
+  return <NativeTableContainer {...props}>{props?.children}</NativeTableContainer>;
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTableContainer.js]: Can be self closing

File: package/components/dataDisplay/CoreTableFooter.js [hunk 4]
```diff
@@ -4,6 +4,24 @@ import React from "react";
 // eslint-disable-next-line import/no-unresolved
 import { NativeTableFooter } from "@wrappid/native";
 
+import { sanitizeComponentProps } from "../../utils/componentUtil";
+
 export default function CoreTableFooter(props) {
-  return <NativeTableFooter {...props}>{props.children}</NativeTableFooter>;
+  props = sanitizeComponentProps(CoreTableFooter, props);
+  return <NativeTableFooter {...props}>{props?.children}</NativeTableFooter>;
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTableFooter.js]: Can be self closing

File: package/components/dataDisplay/CoreTablePagination.js [hunk 5]
```diff
@@ -4,9 +4,12 @@ import React from "react";
...
 import { NativeTablePagination } from "@wrappid/native";
 
+import CoreTableCell from "./CoreTableCell";
 import { DATA_TABLE_CONST } from "../../config/dataTableConstants";
+import { sanitizeComponentProps } from "../../utils/componentUtil";
 
 export default function CoreTablePagination(props) {
+  props = sanitizeComponentProps(CoreTablePagination, props);
   return (
     <NativeTablePagination
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTablePagination.js:22]: Need to remove and check component div

**Review by techoneel — CHANGES_REQUESTED** [review]
Please do the following changes:

1. Remove unnecessary destructure
2. Self closing tag
3. component default value can be removed if problem appears then put it inside default value

File: package/components/dataDisplay/CoreTable.js [hunk 6]
```diff
@@ -4,33 +4,13 @@ import React from "react";
 
 import { NativeTable } from "@wrappid/native";
 
-import { getUUID } from "../../utils/appUtils";
 import { sanitizeComponentProps } from "../../utils/componentUtil";
 
 export default function CoreTable(props) {
-  let _uuid = getUUID();
```
Comment (techoneel) [inline package/components/dataDisplay/CoreTable.js:11]: do not remove this code else comment it for future use or create a todo

**Review by techoneel — DISMISSED** [review]
please do the needed changes or added todo

File: package/components/dataDisplay/CoreTable.js (hunk 6 above)
Comment (anantakumarghosh) [reply package/components/dataDisplay/CoreTable.js:11]: > do not remove this code else comment it for future use or create a todo

This code portion was written for over an year. Removed as containerID was not being used anywhere, and there is no "todo" or any comments for reference.

Please tell me why the below is required and what is the "todo"???

```  let _uuid = getUUID();
  // eslint-disable-next-line no-unused-vars
  let containerId = props?.coreId ? "tc_" + props.coreId : "tc_" + _uuid;
  let _uuid = getUUID();
  // eslint-disable-next-line no-unused-vars
  let containerId = props?.coreId ? "tc_" + props.coreId : "tc_" + _uuid;
```

**Review by techoneel — APPROVED** [review]
good to go

**Pull request merged**